### PR TITLE
Chore: Fix various spelling errors in back-end code

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Subset of #24433, to fix spelling and e.g. make some minor improvements to tests, where I'm more or less certain there won't be any functional changes.

I'd like feedback from team mates to ensure nothing breaks from the changes.

### CODE REVIEW

File: pkg/services/alerting/eval_handler_test.go [hunk 1]
```diff
@@ -153,7 +153,8 @@ func TestAlertingEvaluationHandler(t *testing.T) {
 			So(context.ConditionEvals, ShouldEqual, "[[false OR false] OR false] = false")
 		})
 
-		Convey("Should retuasdfrn no data if one condition has nodata", func() {
+		// FIXME: What should the actual test case name be here?
+		Convey("Should retuasdfrn no data if one condition has no data", func() {
```
Comment (papagian) [inline pkg/services/alerting/eval_handler_test.go:157]:
```suggestion
		Convey("Should not return no data if no condition has no data and using OR", func() {
```

**Review by papagian — CHANGES_REQUESTED** [review]
Generally LGTM, I think there should be a breaking change note for the renamed exported functions.

File: pkg/services/alerting/eval_handler_test.go (hunk 1 above)
Comment (aknuds1) [reply pkg/services/alerting/eval_handler_test.go:157]: @papagian Could we rewrite this more naturally as "Should return data if all conditions have data and using OR"? Good job on figuring out what this test does BTW!?

**Comment by aknuds1** [comment]
> Generally LGTM, I think there should be a breaking change note for the renamed exported functions.

@papagian Do we expose these functions as public API (i.e. not over HTTP)? I wasn't aware of it.

File: pkg/services/alerting/eval_handler_test.go (hunk 1 above)
Comment (papagian) [reply pkg/services/alerting/eval_handler_test.go:157]: yes, that's better

Comment (aknuds1) [reply pkg/services/alerting/eval_handler_test.go:157]: @papagian Please just update the suggestion and I'll commit it :+1:

Comment (papagian) [reply pkg/services/alerting/eval_handler_test.go:157]: on second thought though `context.NoDataFound` is boolean so it does not return data anyway

**Comment by papagian** [comment]
> > Generally LGTM, I think there should be a breaking change note for the renamed exported functions.
> 
> @papagian Do we expose these functions as public API (i.e. not over HTTP)? I wasn't aware of it.

I don't think so but still someone could have used them, no?

File: pkg/services/alerting/eval_handler_test.go (hunk 1 above)
Comment (aknuds1) [reply pkg/services/alerting/eval_handler_test.go:157]: @papagian In that case, maybe we should rephrase it as "Should not return NoData if all conditions have data and using OR"?

File: pkg/services/alerting/eval_handler_test.go [hunk 2]
```diff
@@ -166,7 +167,7 @@ func TestAlertingEvaluationHandler(t *testing.T) {
 			So(context.NoDataFound, ShouldBeFalse)
 		})
 
-		Convey("Should return no data if one condition has nodata", func() {
+		Convey("Should return no data if one condition has no data", func() {
```
Comment (aknuds1) [inline pkg/services/alerting/eval_handler_test.go:170]:
```suggestion
		Convey("Should return NoDataFound if one condition has no data", func() {
```

**Comment by aknuds1** [comment]
> > > Generally LGTM, I think there should be a breaking change note for the renamed exported functions.
> > 
> > 
> > @papagian Do we expose these functions as public API (i.e. not over HTTP)? I wasn't aware of it.
> 
> I don't think so but still someone could have used them, no?

@papagian I think we need to discuss amongst ourselves if we should consider Grafana as exposing a public Go library/API or not. If we don't, as I suspect, we shouldn't carry any responsibility with regards to what's exported.

**Comment by bergquist** [comment]
> > > Generally LGTM, I think there should be a breaking change note for the renamed exported functions.
> > 
> > 
> > @papagian Do we expose these functions as public API (i.e. not over HTTP)? I wasn't aware of it.
> 
> I don't think so but still someone could have used them, no?

@papagian I think we need to discuss amongst ourselves if we should consider Grafana as exposing a public Go library/API or not. If we don't, as I suspect, we shouldn't carry any responsibility with regards to what's exported. The smaller the interface we expose the better.

> > > > Generally LGTM, I think there should be a breaking change note for the renamed exported functions.
> > > 
> > > 
> > > @papagian Do we expose these functions as public API (i.e. not over HTTP)? I wasn't aware of it.
> > 
> > 
> > I don't think so but still someone could have used them, no?
> 
> @papagian I think we need to discuss amongst ourselves if we should consider Grafana as exposing a public Go library/API or not. If we don't, as I suspect, we shouldn't carry any responsibility with regards to what's exported. The smaller the interface we expose the better.

I think we can default to not care about breaking changes for things we export in packages within this project. It's a considerable thought but locking down API's have a high maintenance cost. Perhaps we should make it clear somewhere that depending on code from `grafana/grafana` is at own risk due to breaking changes. 

I cant totally see us making certain packages (dto's and maybe models) considered for external use. in which case we would have to test and think about breaking changes.

**Comment by aknuds1** [comment]
I agree with @bergquist. We might want to provide a public library in the future, but it should have a well defined public interface.

**Review by papagian — APPROVED** [review]

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**